framework/src/meshgenerators/MeshGenerator.C

```diff
@@ -223,15 +237,19 @@ MeshGenerator::buildDistributedMesh(unsigned int dim)
 }
 
 std::unique_ptr<MeshBase>
-MeshGenerator::generateInternal()
+MeshGenerator::generateInternal(const bool data_only)
 {
   libmesh_parallel_only(comm());
   mooseAssert(comm().verify(type() + name()), "Inconsistent execution ordering");
 
-#ifndef NDEBUG
-  for (const auto & [name, mesh] : _requested_meshes)
-    mooseAssert(*mesh, "Null output from " + name);
-#endif
+  if (data_only && !hasGenerateData())
```

Comment:
this check is duplicated in MGSystem ? can you hit both?

Reply:
@loganharbour please resolve this comment once it is addressed in #26296